pkg/storage/mem/mem_store.go

```diff
@@ -127,19 +128,49 @@ func (s *memStore) Get(k string) ([]byte, error) {
 
 // Iterator returns iterator for the latest snapshot of the underlying db.
 func (s *memStore) Iterator(start, limit string) storage.StoreIterator {
-	// TODO Change Store Iterator https://github.com/hyperledger/aries-framework-go/issues/852
+	if start == "" {
```

Comment:
just wondering if `limit` may also need to be checked

Reply:
Fixed. It supposed to be a `limit` not a `start`. Added common tests for this logic.